pkg/controller/manager/manager_controller.go

```diff
@@ -554,6 +554,17 @@ func (r *ReconcileManager) Reconcile(ctx context.Context, request reconcile.Requ
 				r.status.SetDegraded(operatorv1.ResourceCreateError, "Unable to create the tunnel secret", err, logc)
 				return reconcile.Result{}, err
 			}
+		} else {
+			// Check controller references and remove any old APIServer ownership, since ownership of this resource has moved
+			// to the manager controller instead. Without this, we will hit an error when trying to update the secret as it will
+			// have two controllers set.
+			for i := 0; i < len(tunnelCASecret.OwnerReferences); i++ {
+				ref := tunnelCASecret.OwnerReferences[i]
+				if ref.Kind == "APIServer" && ref.Controller != nil && *ref.Controller {
+					tunnelCASecret.OwnerReferences = append(tunnelCASecret.OwnerReferences[:i], tunnelCASecret.OwnerReferences[i+1:]...)
```

Comment:
Wouldn't it be more correct to remove any reference that isn't the manager, or find the manager reference and overwrite the owner reference if it's found (or set it to empty if it wasn't found)? Technically this code would still allow for multiple owners to be set, it's just banking on the fact that there shouldn't be any other owner other than the API.

Reply:
@Brian-McM If there's another owner that doesn't match the condition laid out here, then I want it to fail. Because this is the only known scenario that we want to upgrade from.

Other cases would be something unexpected, so I'm OK with not handling those.

Reply:
Gotcha, sounds reasonable.